ci/scripts/java_jni_build.sh

```diff
@@ -24,6 +24,18 @@ arrow_install_dir=${2}
 build_dir=${3}/java_jni
 # The directory where the final binaries will be stored when scripts finish
 dist_dir=${4}
+normalized_arch=$(arch)
+case ${normalized_arch} in
+  aarch64)
+    normalized_arch=aarch_64
+    ;;
+  i386)
+    normalized_arch=x86_64
+    ;;
+  arm64)
+    normalized_arch=aarch_64
+    ;;
+  esac
```

Comment:
Can we do this in `java/CMakeLists.txt` instead of specifying this by a user?
For example, can we use `CMAKE_HOST_SYSTEM_PROCESSOR` https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html for this?

Reply:
I used CMAKE_HOST_SYSTEM_PROCESSOR now.